### DISCUSSION TITLE
Options Picker for Number Field

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#7283] [Help]
I have MySQL datasource with a column (Report_Year) of Number type:
![image](https://user-images.githubusercontent.com/106758851/184705840-6e091063-0b73-46ff-8046-825cb6af0ba7.png)

In the form field components it has this a number field input. I added a separate Options Picker component and tried to set the Field to this column, but noticed that only text columns can be linked to the Options Picker. Is this something that can be changed somehow that the options picker can be a list of numbers to select from?

![image](https://user-images.githubusercontent.com/106758851/184706755-fb6717f1-780f-4bcf-ac27-f31effcc4192.png)

I would generate a list of years for the options by using javascript and the options picker should look similar to this:

![image](https://user-images.githubusercontent.com/106758851/184707121-221136ae-8f9d-4bac-985e-255b8f2323d8.png)

**Comment by melohagan** [comment]
Hey @mmalessa80 

You need to use the *Data Provider* method of providing options: https://docs.budibase.com/docs/option-picker#using-a-data-provider

Then you can type the name of the column into the field name. Even though it doesn't appear in the dropdown, the save will still work correctly.

![Screenshot 2022-08-18 at 10 22 56](https://user-images.githubusercontent.com/101575380/185360294-fb105bb2-940e-474c-864b-b66934a3ad64.png)

![Screenshot 2022-08-18 at 10 22 00](https://user-images.githubusercontent.com/101575380/185360009-36901b97-1416-43e4-884d-40ad0c8c22b4.png)

![Screenshot 2022-08-18 at 10 22 36](https://user-images.githubusercontent.com/101575380/185360187-e37fc94a-79d3-47ed-a46c-299d993d4685.png)
